stream/src/main/java/org/apache/kafka/streaming/kstream/internals/KStreamJoin.java

```diff
@@ -31,47 +31,23 @@
         abstract Iterator<T> find(K key, long timestamp);
     }
 
-    private final String windowName1;
-    private final String windowName2;
+    private final String windowName;
     private final ValueJoiner<V1, V2, V> joiner;
 
-    private Processor processorForOtherStream = null;
-    public final ProcessorDef processorDefForOtherStream = new ProcessorDef() {
-        @Override
-        public Processor instance() {
-            return processorForOtherStream;
-        }
-    };
-
-    KStreamJoin(String windowName1, String windowName2, ValueJoiner<V1, V2, V> joiner) {
-        this.windowName1 = windowName1;
-        this.windowName2 = windowName2;
+    KStreamJoin(String windowName, ValueJoiner<V1, V2, V> joiner) {
```

Comment:
Be careful with rebase. Rebase is tricky. Usually you don't need to use it because git is smart. I always pull from the latest, and it cleans up unwanted diffs.